### PR TITLE
Separate testing from prod

### PR DESCRIPTION
**Pull request by coconutpalm**
- Separate testing dependencies from prod dependencies
- Minor code cleanups
- Updated README.md

### CODE REVIEW

File: src/clj_foundation/templates.clj [hunk 1]
```diff
@@ -130,7 +130,7 @@
 
   If :partial-resolve true is specified in options, resolves as many
   variables as possible.  Unresolved variables are returned as ${var-name}."
-  [parameter-source :- (types String clojure.lang.ISeq)
+  [parameter-source :- (types String clojure.lang.ASeq)
```
Comment (coconutpalm) [inline src/clj_foundation/templates.clj:133]: The (types... schema needs concrete types.  (instance? IFoo val) isn't sufficient.

File: test/clj_foundation/config_test.clj [hunk 2]
```diff
@@ -2,9 +2,10 @@
   (:require [clojure.test :refer :all]
             [clojure.java.io :as io]
             [schema.core :as s :refer [=> =>*]]
+            [clj-foundation.unit-test-common :as common]
             [clj-foundation.config :refer :all]))
 
-(s/set-compile-fn-validation! true)
+(common/register-fixtures)
```
Comment (coconutpalm) [inline test/clj_foundation/config_test.clj:9]: Migrated all test namespaces to (common/register-fixtures) rather than ad hoc initialization as before.

**Comment by gsavovski** [comment]
👍

**Comment by baguthrie** [comment]
👍

File: test/clj_foundation/templates_test.clj [hunk 3]
```diff
@@ -2,10 +2,12 @@
   (:require [clojure.test :refer :all]
             [clojure.java.io :as io]
             [schema.core :as s :refer [=> =>*]]
+            [clj-foundation.unit-test-common :as common]
             [clj-foundation.templates :refer :all])
   (:import [clojure.lang ExceptionInfo]))
 
-(s/set-compile-fn-validation! true)
+
+(common/register-fixtures)
```
Comment (recursive-automata) [inline test/clj_foundation/templates_test.clj:10]: Well factored. :)

**Comment by recursive-automata** [comment]
👍

**Pull request merged**